backend-api/src/functions/adapters/kmsAdapter.ts

```diff
@@ -1,50 +1,70 @@
 import {
   DecryptCommand,
   DecryptCommandOutput,
+  IncorrectKeyException,
+  InvalidCiphertextException,
   KMSClient,
 } from "@aws-sdk/client-kms";
-import { NodeHttpHandler } from "@smithy/node-http-handler";
 import { errorResult, Result, successResult } from "../utils/result";
+import { NodeHttpHandler } from "@aws-sdk/node-http-handler";
 
-export class KMSAdapter {
-  private readonly encryptionKeyArn: string;
-
-  constructor(encryptionKeyArn: string) {
-    this.encryptionKeyArn = encryptionKeyArn;
-  }
+export class KMSAdapter implements IKmsAdapter {
+  private readonly kmsClient = new KMSClient([
+    {
+      region: "eu-west-2",
+      maxAttempts: 2,
+      requestHandler: new NodeHttpHandler({
+        connectionTimeout: 29000,
+        requestTimeout: 29000,
+      }),
+    },
+  ]);
 
   async decrypt(
-    encryptedCek: Uint8Array,
-  ): Promise<Result<DecryptCommandOutput>> {
-    let output: DecryptCommandOutput;
+    keyArn: string,
+    ciphertext: Uint8Array,
+  ): Promise<Result<Uint8Array>> {
+    let decryptCommandOutput: DecryptCommandOutput;
     try {
-      output = await KmsClient.send(
+      decryptCommandOutput = await this.kmsClient.send(
         new DecryptCommand({
-          KeyId: this.encryptionKeyArn,
-          CiphertextBlob: encryptedCek,
+          KeyId: keyArn,
+          CiphertextBlob: ciphertext,
           EncryptionAlgorithm: "RSAES_OAEP_SHA_256",
         }),
       );
-    } catch {
+    } catch (error) {
+      if (
+        error instanceof InvalidCiphertextException ||
```

Comment:
I don't think we can use the exception property `$fault` to decide whether we return a client or a server error - we can talk about this over a call - so we need to check the exception type. I think we need to have a discussion about how we want to handle the different exceptions KMS can throw and if we want to return CLIENT_ERROR or something else from the function.

Reply:
Yes. When we call KMS, we are the client and they are the server, so the fault property (when set to `client`) is saying that we as their client made a mistake, but that does not mean that our client made a mistake. 🫤

Reply:
Re your first question, the KMS `decrypt` method throws various exceptions. We don't need to check every single one of them in the code but we should probably go through the list of possible exceptions and decide which ones should be a client error and which should be server error. Or maybe we don't want to check at all and just return a server error!

Reply:
Yeah I'm thinking a blanket server error actually might be the way to go here. I'm trying to think of the value of checking but can't think of much

Reply:
I've stopped checking the KMS exception type and instead chose to catch all errors and return a SERVER_ERROR like before just so that we can merge this PR. However, I do think we need to have a conversation about error handling because right now I feel that we are swallowing these errors and we can't tell what's going wrong. :confusing